generator/PythonGenerator.py

```diff
@@ -0,0 +1,622 @@
+                "PropertyValueList",
+                "Protocol",
+                "ExperimentalActivity",
+                "ProtocolExecution",
+                "QuantitativeValue",
+                "QuantitativeValueRange",
+                "QuantitativeValueArray",
+                "ServiceLink",
+                "StringParameter",
+                "Subject",
+                "SubjectGroup",
+                "SubjectGroupState",
+                "SubjectState",
+                "TissueSample",
+                "TissueSampleCollection",
+                "TissueSampleCollectionState",
+                "TissueSampleState",
+                "BehavioralProtocol",
+                "Stimulation",
+                "Strain"
+            ],
+            "model": ["Model", "ModelVersion"],
+            "software": ["SWHID", "Software", "SoftwareVersion"],
+            "restricted": ["ContactInformation"],
+            "metadatamodel": ["MetaDataModel", "MetaDataModelVersion"],
+            "controlled": ["License", "ContentType"]
+        }),
+    "computation": {
+        "default": "computation"
+    },
+    "controlledTerms": {
+        "default": "controlled"
+    },
+    "SANDS":
+    invert_dict(
+        {
+            "spatial": [
+                "AnatomicalEntity",
+                "Annotation",
+                "CoordinatePoint",
+                "CustomAnatomicalEntity",
+                "CustomAnnotation",
+                "CustomCoordinateSpace",
+                "Image",
+                "QualitativeRelationAssessment",
+                "QuantitativeRelationAssessment"
+            ],
+            "atlas": [
+                "AnatomicalTargetPosition",
+                "AtlasAnnotation",
+                "BrainAtlas",
+                "BrainAtlasVersion",
+                "Circle",
+                "CommonCoordinateSpace",
+                "CoordinatePoint",
+                "Ellipse",
+                "ParcellationEntity",
+                "ParcellationTerminology",
+                "ParcellationTerminologyVersion",
+                "ParcellationEntityVersion",
+                "Rectangle"
+            ]
+        }),
+    "publications": {
+        "default": "publications"
+    },
+    "ephys": {
+        "default": "electrophysiology"
+    }
+}
+
+
+def get_default_space(schema_group, cls_name):
+    print("get_default_space")
+    if cls_name in DEFAULT_SPACES[schema_group]:
+        print(DEFAULT_SPACES[schema_group][cls_name])
+        return DEFAULT_SPACES[schema_group][cls_name]
+    else:
+        print("No default space")
+        print(cls_name)
+        print(schema_group)
+        print(DEFAULT_SPACES[schema_group])
+        return DEFAULT_SPACES[schema_group]["default"]
+
+
+# in general, we use the required fields when deciding whether a given object already exists
+# in the KG. Sometimes this method is inappropriate or undesired, and so for some classes
+# we use a custom set of fields.
+custom_existence_queries = {
```

Comment:
this is also KG-specific, although we could discuss whether this kind of information could be included in openMINDS.

Reply:
Good point, I removed this.